subsys/nrf_security/src/drivers/cracen/cracenpsa/src/ikg_signature.c

```diff
@@ -0,0 +1,199 @@
+/* Isolated key operations (signature generation, ...)
+ *
+ * Copyright (c) 2025 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ *
+ * Workmem layout for the sign operation:
+ *      1. Hash digest of the message to be signed (size: digestsz).
+ *
+ * Workmem layout for the sign digest operation:
+ *      1. Hash digest of the message to be signed (size: digestsz).
+ *
+ * Other IKG tasks don't need workmem memory.
+ */
+
+#include <stdint.h>
+#include <string.h>
+#include <silexpk/core.h>
+#include <silexpk/iomem.h>
+#include <sxsymcrypt/hash.h>
+#include <sxsymcrypt/hashdefs.h>
+#include <silexpk/ik.h>
+#include <cracen/statuscodes.h>
+#include <cracen_psa_ikg.h>
+#include <cracen_psa_primitives.h>
+#include "common.h"
+
+#define MAX_ECDSA_ATTEMPTS 255
+
+/** Convert a digest into an operand for ECDSA
+ *
+ * The raw digest may need to be padded or truncated to fit the curve
+ * operand used for ECDSA.
+ *
+ * Conversion could also imply byte order inversion, but that is not
+ * implemented here. It's expected that SilexPK takes big endian
+ * operands here.nnn
+ *
+ * As the digest size is expressed in bytes, this procedure does not
+ * work for curves which have sizes not multiples of 8 bits.
+ */
+static void digest2op(const uint8_t *digest, size_t sz, uint8_t *dst, size_t opsz)
+{
+	if (opsz > sz) {
+		sx_clrpkmem(dst, opsz - sz);
+		dst += opsz - sz;
+	}
+	sx_wrpkmem(dst, digest, opsz);
+}
+
+static void ecdsa_read_sig(struct ecdsa_signature *sig, const uint8_t *r, const uint8_t *s,
+			   size_t opsz)
+{
+	sx_rdpkmem(sig->r, r, opsz);
+	if (!sig->s) {
+		sig->s = sig->r + opsz;
+	}
+	sx_rdpkmem(sig->s, s, opsz);
+}
+
+static const struct sxhashalg *select_optimal_hashalg(void)
+{
+	const struct sx_pk_capabilities *caps = sx_pk_fetch_capabilities();
+	int sz = caps->ik_opsz;
```

Comment:
I know that you copied this function so its not necessary to change it here. But I think the switch case here is basically useless. We can just remove the function all together and just hardcode the hash algorithm that is supported in the relevant functions.

Reply:
It does indeed do nothing currently, or rather it always picks the same choice. My reasoning for leaving it was if there was a variant of CRACEN in the future with a different key type/hash then it could be useful, but I don't mind removing it

Reply:
I do prefer it removed but this can be done later. So completely up to you what you want to do.